WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1775,19 +1780,23 @@ public boolean updatePostContentNewEditor(boolean isAutoSave, String title, Stri
         }
 
         boolean titleChanged = PostUtils.updatePostTitleIfDifferent(mPost, title);
-        boolean contentChanged;
-        if (mEditorFragment instanceof AztecEditorFragment) {
-            contentChanged = ((AztecEditorFragment)mEditorFragment).hasHistory();
-            if (contentChanged) {
-                mPost.setContent(content);
-            } else if (!((AztecEditorFragment)mEditorFragment).isHistoryEnabled()){
-                // if history is not enabled, then we can only confirm whether there's been a content change
-                // by comparing content
+        boolean contentChanged = compareCurrentMediaMarkedUploadingToOriginal(content);
```

Comment:
It feels like this whole block could be simplified to:
```
boolean contentChanged;
if (compareCurrentMediaMarkedUploadingToOriginal(content)) {
    contentChanged = true;
} else if (mEditorFragment instanceof AztecEditorFragment 
        && ((AztecEditorFragment) mEditorFragment).isHistoryEnabled()) {
    contentChanged = ((AztecEditorFragment) mEditorFragment).hasHistory();
} else {
    contentChanged = mPost.getContent().compareTo(content) != 0;
}
if (contentChanged) {
    mPost.setContent(content);
}
```

Reply:
Excellent - just tested it in all 3 cases (images, video, and just user changes through the editor). Replaced the code with your block in ccf116b